### PR TITLE
http: fix signal slot headers with non-batching but prefix-suffix case

### PR DESCRIPTION
**Pull request by furiel**
In case batching is not enabled, but prefix and suffix is used, http-worker (`_insert_single`) calculated headers before the body is finshed (body-suffix is added).

After this patch, the msg specific headers are added in `_insert_single`, and everything else during flush, unconditionally.

No need for changenote, just a small addition to #3103.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/http/tests/test_http-signal_slot.c [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  SignalSlotConnector *ssc = driver->super.super.super.super.signal_slot_connector;
+
+  const gchar *test_msg = "test message";
+  CONNECT(ssc, signal_http_header_request, _check, test_msg);
+
+  cr_assert(log_pipe_init((LogPipe *)driver));
+
+  _generate_message(driver, test_msg);
+
+  wait_for_signal_slot_to_finish();
```
Comment (lbudai) [inline modules/http/tests/test_http-signal_slot.c]: @furiel : signal-slot mechanism is _synchronous_ in syslog-ng.

File: modules/http/http-worker.c [hunk 2]
```diff
@@ -605,9 +602,7 @@ _flush(LogThreadedDestWorker *s, LogThreadedFlushMode mode)
     return LTR_RETRY;
 
   _finish_request_body(self);
-
-  if (list_is_empty(self->request_headers))
```
Comment (lbudai) [inline modules/http/http-worker.c]: why don't we need this check?

File: modules/http/tests/test_http-signal_slot.c (hunk 1 above)
Comment (lbudai) [reply modules/http/tests/test_http-signal_slot.c]: as @furiel  pointed out: we really start another thread...
I've suggested to use `cond_wait` instead : wait in the caller, and cond_signal in the slot... 
But it's up to you @furiel  :)

Comment (furiel) [reply modules/http/tests/test_http-signal_slot.c]: You are right. cond_wait is better. I will fix this next time.

**Pull request merged**